src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -162,7 +166,9 @@ private static double Hypot(double a, double b)
         private static double Log1P(double x)
         {
             // Compute log(1 + x) without loss of accuracy when x is small.
+
             // Our only use case so far is for positive values, so this isn't coded to handle negative values.
+            Debug.Assert(!(x < 0.0));
```

Comment:
Agreed. There is a reason it's written like this, but you won't like it: it's so that NaN's won't violate the assert. One way to handle this is to add special NaN handling before we get to the assert, but I really don't want to add release code just to make debug asserts prettier. So for now I am changing these to Debug.Assert((x >= 0.0) || Double.IsNaN(x)).